### PR TITLE
tuxedo-keyboard: 2019-08-26 -> v3.0.5

### PR DESCRIPTION
**Pull request by blanky0230**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/119658

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/tuxedo-keyboard/default.nix [hunk 1]
```diff
@@ -2,16 +2,17 @@
 
 stdenv.mkDerivation rec {
   pname = "tuxedo-keyboard-${kernel.version}";
-  version = "2019-08-26";
+  version = "v3.0.5";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/tuxedo-keyboard/default.nix:5]:
```suggestion
  version = "3.0.5";
```

File: pkgs/os-specific/linux/tuxedo-keyboard/default.nix [hunk 2]
```diff
@@ -2,16 +2,17 @@
...
 
   src = fetchFromGitHub {
     owner = "tuxedocomputers";
     repo = "tuxedo-keyboard";
-    rev = "d65e76e84cfd8169591fc2a0a7c9219fa19da1b5";
-    sha256 = "1s48qpwybwh5pwqas2d1v2a7x4r97sm4hr9i4902r1d7h384bv17";
+    rev = "v3.0.5";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/tuxedo-keyboard/default.nix:10]:
```suggestion
    rev = "v${version}";
```

We do not want to repeat the version number to make updates easier.

File: pkgs/os-specific/linux/tuxedo-keyboard/default.nix [hunk 3]
```diff
@@ -2,16 +2,17 @@
...
+    sha256 = "123ady2bi2dwbajy3pgv10l3g2pyhi5k31c1ii0zcrvl2qqhndck";
   };
 
-  makeFlags = [ "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build" ];
+  makeFlags =
+    [ "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build" ];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/tuxedo-keyboard/default.nix:15]:
```suggestion
  makeFlags = [ "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build" ];
```

File: pkgs/os-specific/linux/tuxedo-keyboard/default.nix [hunk 4]
```diff
@@ -21,7 +22,7 @@ stdenv.mkDerivation rec {
   meta = with lib; {
     description = "Full color keyboard driver for tuxedo computers laptops";
     homepage = "https://github.com/tuxedocomputers/tuxedo-keyboard/";
-    license = licenses.gpl2;
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/tuxedo-keyboard/default.nix:25]: Is that gpl3Only or gpl3Plus?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121266` at 435fff83 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.tuxedo-keyboard</li>
    <li>linuxPackages_hardkernel_4_14.tuxedo-keyboard</li>
    <li>linuxPackages_hardkernel_latest.tuxedo-keyboard</li>
    <li>linuxPackages_latest-libre.tuxedo-keyboard</li>
    <li>linuxPackages_testing_bcachefs.tuxedo-keyboard</li>
    <li>linuxPackages_xanmod.tuxedo-keyboard</li>
  </ul>
</details>
<details>
  <summary>11 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/3444bad1f47d03e8b463d6f749766e5c">linuxPackages.tuxedo-keyboard</a> (linuxPackages_5_10.tuxedo-keyboard)</li>
    <li><a href="https://gist.github.com/7512d10251a57fb5adad28a39cdea835">linuxPackages_4_14.tuxedo-keyboard</a></li>
    <li><a href="https://gist.github.com/af8d4dd7c6ca16c7a29d562807915883">linuxPackages_4_19.tuxedo-keyboard</a></li>
    <li><a href="https://gist.github.com/1e4df6c5cd2bdb006759cccc0cbbe348">linuxPackages_4_4.tuxedo-keyboard</a></li>
    <li><a href="https://gist.github.com/98b280a6cfa6af5292edd1394a7aae15">linuxPackages_4_9.tuxedo-keyboard</a></li>
    <li><a href="https://gist.github.com/ae25629311d011e9ffb2bcf7961ce10e">linuxPackages_5_11.tuxedo-keyboard</a> (linuxPackages_latest.tuxedo-keyboard)</li>
    <li><a href="https://gist.github.com/84f8b741fa4827501b74d0e095c9cbe3">linuxPackages_5_4.tuxedo-keyboard</a></li>
    <li><a href="https://gist.github.com/39e199fcd5c9dfbd30712fab31bc315f">linuxPackages_hardened.tuxedo-keyboard</a></li>
    <li><a href="https://gist.github.com/bb92b0dbf2e2fa47ae6687ba7004dbf9">linuxPackages_latest_hardened.tuxedo-keyboard</a></li>
    <li><a href="https://gist.github.com/80e3e27f7f68a1ac1f0bc4e0ad146ca3">linuxPackages_lqx.tuxedo-keyboard</a></li>
    <li><a href="https://gist.github.com/eaa718d123991bc7d3210c3d4066714c">linuxPackages_zen.tuxedo-keyboard</a></li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/os-specific/linux/tuxedo-keyboard/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/tuxedo-keyboard/default.nix:25:5:
```
   |
25 |     license = licenses.gpl3;
   |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `kmod`.
Near pkgs/os-specific/linux/tuxedo-keyboard/default.nix:1:41:
```
  |
1 | { lib, stdenv, fetchFromGitHub, kernel, kmod }:
  |                                         ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

File: pkgs/os-specific/linux/tuxedo-keyboard/default.nix (hunk 4 above)
Comment (blanky0230) [reply pkgs/os-specific/linux/tuxedo-keyboard/default.nix]: gpl3Plus :) Thanks for pointing it out. Completely forgot about that.

File: pkgs/os-specific/linux/tuxedo-keyboard/default.nix [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
   pname = "tuxedo-keyboard-${kernel.version}";
-  version = "2019-08-26";
+  version = "3.0.5";
 
   src = fetchFromGitHub {
     owner = "tuxedocomputers";
     repo = "tuxedo-keyboard";
-    rev = "d65e76e84cfd8169591fc2a0a7c9219fa19da1b5";
-    sha256 = "1s48qpwybwh5pwqas2d1v2a7x4r97sm4hr9i4902r1d7h384bv17";
+    rev = "${version}";
```
Comment (Ma27) [inline pkgs/os-specific/linux/tuxedo-keyboard/default.nix:10]: AFAIU this should be `v${version}`. FWIW it's currently failing to download the tarball for me:

```
λ ma27 [~/Projects/nixpkgs] at  tuxedo-keyboard-update ✔
→ b linuxPackages.tuxedo-keyboard                                                           [a872c3b0f2b]
source> trying https://github.com/tuxedocomputers/tuxedo-keyboard/archive/3.0.5.tar.gz
source>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
source>                                  Dload  Upload   Total   Spent    Left  Speed
source> 100   138  100   138    0     0    552      0 --:--:-- --:--:-- --:--:--   552
source>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
source> curl: (22) The requested URL returned error: 404
source> error: cannot download source from any mirror
error: builder for '/nix/store/1z88pcvb6vzx0r4xzwfsndls8f7jaxkf-source.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/7gjzybhbrxx2yd73nch13j5rviw0zm4j-tuxedo-keyboard-5.10.32-3.0.5.drv' failed to build

```

Comment (blanky0230) [reply pkgs/os-specific/linux/tuxedo-keyboard/default.nix]: :facepalm: thanks! :)

**Comment by blanky0230** [comment]
resolved https://github.com/NixOS/nixpkgs/issues/119658

**Comment by Ma27** [comment]
Oh right :) Next time you may want to add something like `Closes #ticketnum` to your commit message's body, then GitHub will take care of this as soon as somebody merges :)

**Pull request merged**